.github/workflows/build.yml

```diff
@@ -3,6 +3,14 @@ on:
       inputs:
         amplify-app-id:
           description: "The ID for your Amplify app (which you can get from the end of its ARN)"
+          required: false
```

Comment:
How come the app id and some other things like the bucket name are no longer required?

Reply:
Ah I guess it's because we're not always deploying

Reply:
In its current state, we wouldn't be able to reuse `build.yml` for targeting different node versions / OS combos, as it'd attempt to do a deploy for both.

Ideally, `build` & `deploy` would be their own jobs, but given we don't have anything like turbo, I wanted to avoid caching the build `dists` or storing them as `build` artifacts.

By having the deploy functionality within the same job, it means we can't avoid this.


The downside to this, is that I've had to refactor the code with a new boolean `deploy` input. These AWS / Amplify inputs will only ever be used if `deploy` is set to true.

Annoyingly, GitHub doesn't let you set the required property conditionally, otherwise we could do something like:
`required: ${{ inputs.deploy == true }}` 

I hope this makes sense. It's not ideal, and probably could do with a rework in the future, but it was the simplest option for now without introducing a load of complexity.